src/test/scala-2.11/com/linagora/gatling/imap/scenario/ImapFullScenario.scala

```diff
@@ -0,0 +1,67 @@
+package com.linagora.gatling.imap.scenario
+
+import java.util.Calendar
+
+import com.linagora.gatling.imap.PreDef._
+import com.linagora.gatling.imap.protocol.Uid
+import com.linagora.gatling.imap.protocol.command.FetchAttributes.AttributeList
+import com.linagora.gatling.imap.protocol.command.FetchRange._
+import io.gatling.app.Gatling
+import io.gatling.core.Predef._
+import io.gatling.core.config.GatlingPropertiesBuilder
+import io.gatling.core.scenario.Simulation
+
+import scala.collection.immutable.Seq
+import scala.concurrent.duration._
+
+class ImapFullScenario extends Simulation {
+  val feeder = csv("users.csv").circular
+  val userCount: Int = feeder.records.length
+
+  val receiveEmail = exec(imap("append").append("INBOX", Some(Seq("\\Flagged")), Option.empty[Calendar],
+                  """From: expeditor@example.com
+                    |To: recipient@example.com
+                    |Subject: test subject
+                    |
+                    |Test content
+                    |abcdefghijklmnopqrstuvwxyz
+                    |0123456789""".stripMargin).check(ok))
+
+  val readLastEmail = exec(imap("list").list("", "*").check(ok, hasFolder("INBOX")))
+                      .exec(imap("select").select("INBOX").check(ok))
+                      .exec(imap("fetch").fetch(Seq(Last()), AttributeList("BODY[HEADER]", "UID", "BODY[TEXT]")).check(ok))
+
+  val heavyUser = repeat(3)(receiveEmail)
+                  .repeat(2)(readLastEmail)
+  val lightUser = receiveEmail
+                  .exec(readLastEmail)
+
+  val duration = 4 hours
+
+  val scn = scenario("Imap").feed(feeder)
+    .pause(1 second)
+    .exec(imap("Connect").connect()).exitHereIfFailed
+    .exec(imap("login").login("${username}", "${password}").check(ok))
+    .exec {
+      randomSwitch(
+        75.0 -> during(duration) {
```

Comment:
I do not agree with your pace time (way too long) regarding scenario duration and number of users.

Please divide by 30 :-)